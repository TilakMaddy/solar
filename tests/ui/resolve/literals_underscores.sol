contract LT {
    uint256 a = 1000_; //~ERROR: invalid use of underscores in number literal
    uint256 b = 100__0;//~ERROR: invalid use of underscores in number literal
    uint256 c = 1_.4e10;//~ERROR: invalid use of underscores in number literal
    uint256 d = 3.4_e10;//~ERROR: invalid use of underscores in number literal
    uint256 e = 3.4e_10;//~ERROR: invalid use of underscores in number literal

    // exception
    uint256 f = 3._4e10; // Does not show up
}
