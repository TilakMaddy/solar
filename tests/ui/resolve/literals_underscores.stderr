error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint256 a = 1000_;
   |                 ^^^^^
   |
   = help: remove trailing underscores

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint256 b = 100__0;
   |                 ^^^^^^
   |
   = help: only 1 consecutive underscore `_` is allowed between digits

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint256 c = 1_.4e10;
   |                 ^^^^^^^
   |
   = help: remove underscores in front of the fraction part

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint256 d = 3.4_e10;
   |                 ^^^^^^^
   |
   = help: remove underscores at the end of the mantissa

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint256 e = 3.4e_10;
   |                 ^^^^^^^
   |
   = help: remove underscores in front of the exponent

error: aborting due to 5 previous errors

